test/topology_custom/test_replace_with_same_ip_twice.py

```diff
@@ -0,0 +1,40 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+import time
+from test.pylib.scylla_cluster import ReplaceConfig
+from test.pylib.manager_client import ManagerClient
+from test.topology.util import wait_for_token_ring_and_group0_consistency
+from test.pylib.internal_types import ServerInfo
+import pytest
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.asyncio
+async def test_replace_with_same_ip_twice(manager: ManagerClient) -> None:
+    """Replace an existing node with new node using the same IP address, and then change its IP"""
+
+    logger.info("starting a cluster with two nodes")
+    servers = await manager.servers_add(3, config={'failure_detector_timeout_in_ms': 2000})
+    logger.info(f"cluster started {servers}")
+
+    async def replace_with_same_ip(s: ServerInfo) -> ServerInfo:
+        logger.info(f"stopping server {s.server_id}")
+        await manager.server_stop(s.server_id)
+
+        logger.info(f"replacing server {s.server_id} with same ip")
+        replace_cfg = ReplaceConfig(replaced_id = s.server_id, reuse_ip_addr = True, use_host_id = False)
+        result: ServerInfo = await manager.server_add(replace_cfg)
+
+        logger.info(f"waiting for token ring and group0 consistency")
+        await wait_for_token_ring_and_group0_consistency(manager, time.time() + 30)
```

Comment:
Have you added this after observing a failure as a workaround? It shouldn't be needed in general in raft-topology mode; it was originally intended for gossip-based topology where token ring was eventually consistent and took time to converge. If we still need it, something's not right

Reply:
> I can include another commit and remove wait_for_token_ring_and_group0_consistency from all raft-topology tests.

Meh, I'd say it's out of scope.